lasagne/objectives.py

```diff
@@ -12,74 +14,136 @@ def mse(x, t):
     :returns:
         - output : the mean square error across all dimensions
     """
-    return T.mean((x - t) ** 2)
+    return (x - t) ** 2
 
 
-def crossentropy(x, t):
-    """Calculates the binary crossentropy mean across all dimentions,
-    i.e.  feature dimension AND minibatch dimension.
+class Objective(object):
+    _valid_aggregation = {None, 'mean', 'sum'}
 
-    :parameters:
-        - x : predicted values
-        - t : target values
+    """
+    Training objective
 
-    :returns:
-        - output : the mean binary cross entropy across all dimensions
+    The  `get_loss` method returns an expression that can be used for
+    training with a gradient descent approach.
```

Comment:
That's a little misleading, it's not limited to gradient descent. How about: "The `get_loss` method returns a cost expression useful for training or evaluating a network." ?